src/Inventory/Asset/MainAsset.php

```diff
@@ -143,7 +143,8 @@ public function prepare(): array
             }
 
             //for Computer / Phone get remote_addr from agent
-            if (isset($this->getAgent()->fields['remote_addr'])) {
+            //some TU only use handle() -> $this->getAgent() reutrn null
+            if (!defined('TU_USER') && isset($this->getAgent()->fields['remote_addr'])) {
```

Comment:
If it is possible to not use `TU_USER` outside test suite code, it would be better.
```suggestion
            if ($this->getAgent() !== null && isset($this->getAgent()->fields['remote_addr'])) {
```

Reply:
Problem ```getAgent``` is design to return ```Agent``` 

```php
    /**
     * Get agent
     *
     * @return Agent
     */
    public function getAgent(): Agent
    {
        return $this->agent;
    }
```

Reply:
```suggestion
            if ($this->agent instanceof \Agent && isset($this->getAgent()->fields['remote_addr'])) {
```

Reply:
done